assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -302,7 +304,28 @@ const RightMenu: React.FC<any> = () => {
     },
     [currentActivity, currentPartInstance, currentPartSelection],
   );
-
+  const handleEditComponentJson = (newJson: any, isCancelled = false) => {
+    setDisplayEditor(false);
+    if(isCancelled)
+      return;
+    const cloneActivity = clone(currentActivity);
+    const ogPart = cloneActivity.content?.partsLayout.find((part: any) => part.id === currentPartSelection);
+    if (!ogPart) {
+      console.warn(
+        'couldnt find part in current activity, most like lives on a layer; you need to update they layer copy directly',
+      );
+      return;
+    }
+    if (newJson.id !== ogPart.id) {
```

Comment:
should check that `newJson.id` is also not blank. it would also be good (same with the other component editor) now that I think of it to check all the part id in the `currentActivityTree` to make sure it isn't duplicated. both of those are a bit tricky however since we don't have any other error messages anywhere... at least with this modal we can pass in an array of all the ids I think and then it can use that for validation, displaying an error message and disabling the Save button.